app/models/post.rb

```diff
@@ -0,0 +1,12 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  has_many :comments
+  has_many :likes
+
+  def self.update_posts_counter(user_id)
+    user = User.find(user_id)
+    user.update(posts_counter: user.posts.count)
+  end
```

Comment:
- [ ] [optiona] @Lornakaboro It would have been nice if you used one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_posts_counter method every time a new post has been saved to database, then make the update_posts_counter method private, this also applies to the like and comment models